libraries/lib-wave-track/WaveTrack.cpp

```diff
@@ -187,8 +189,30 @@ WaveTrack::Interval::Interval(const ChannelGroup &group,
 {
 }
 
+WaveTrack::Interval::Interval(
+   const ChannelGroup& group, size_t width,
+   const SampleBlockFactoryPtr& factory, int rate)
+    : Interval(
+         group, std::make_shared<WaveClip>(1, factory, floatSample, rate, 0),
+         width == 2 ?
+            std::make_shared<WaveClip>(1, factory, floatSample, rate, 0) :
+            nullptr)
+{
+}
+
 WaveTrack::Interval::~Interval() = default;
 
+void WaveTrack::Interval::Append(constSamplePtr buffer[], size_t len)
+{
+   for (unsigned channel = 0; channel < NChannels(); ++channel)
+      GetClip(channel)->AppendNewBlock(buffer[channel], floatSample, len);
```

Comment:
More about formats:  either the argument to this function should be `float *const *` or else the format should be an argument, not assumed to be `floatSample`.  The latter solution has the appeal of more generality easily implemented, even if not yet needed.

Reply:
Absolutely - I've been overseeing this argument. Addressing ...

Reply:
"Overseeing" or "overlooking"?  (Not the same...)